Add integration docs for metrics forwarding (#3038)
astronomer/docs: astro/export-datadog.md

:::

1. Create a new Datadog API key or copy an existing API key. See [API and Application Keys](https://docs.datadoghq.com/account_management/api-app-keys/).
2. In the Cloud UI, select a Workspace and then select an Astro Deployment for which you want to export metrics.
3. Create a new [environment variable](environment-variables.md#set-environment-variables-in-the-cloud-ui) in your Deployment with the Datadog API key from step 1:
   
   - **Key:** `DATADOG_API_KEY`
   - **Value:** `<Your-Datadog-API-key>`.
  
   Select the **Secret?** checkbox. This ensures that your Datadog API key is saved securely and is not available to Workspace users in plain text.

4. (Optional) Add the following environment variable if your organization doesn't use the default Datadog site `datadoghq.com`:
   
   - **Key:** `DATADOG_SITE`
   - **Value:** `<Your-Datadog-Site>`
  
5. (Optional) Add the following environment variables to create [custom Datadog tags](https://docs.datadoghq.com/getting_started/tagging/) associated with your Deployment:

>    - **Key 1**: `AIRFLOW__METRICS__STATSD_DATADOG_ENABLED`
>    - **Value 1**: `True`
> 
>   
>     - **Key 2**: `AIRFLOW__METRICS__STATSD_DATADOG_TAGS`
>    - **Value 2**: `<tag-key-1>:<tag-value-1>,<tag-key-2>:<tag-value-2>`

Reviewer: ```suggestion
   - **Key 1**: `AIRFLOW__METRICS__STATSD_DATADOG_ENABLED`
   - **Value 1**: `True`

   - **Key 2**: `AIRFLOW__METRICS__STATSD_DATADOG_TAGS`
   - **Value 2**: `<tag-key-1>:<tag-value-1>,<tag-key-2>:<tag-value-2>`
```
something funky going on with the spacing here

---
sidebar_label: 'Datadog'
title: 'Export metrics and logs to Datadog'
id: export-datadog
description: "Configure your Deployment to forward observability data to your Datadog instance."
---

> Using Datadog with Astro allows you to observe your Deployments using Datadog's extensive cloud monitoring tools. Currently, Astro supports sending the following information to Datadog:

Reviewer: I think we need one more sentence about the "why" between the first and second sentences here. Something about how this means they can add their Astro observability into any existing practices? More easily customize their monitoring and reporting about usage? Consolidate their data analysis in one place?

- Airflow task logs
- Supported Datadog [Airflow metrics](https://docs.datadoghq.com/integrations/airflow/?tab=host#data-collected)

> Complete the following setup to authenticate your Deployments to Astro and forward their observability data to your Datadog instance.

Reviewer: `authenticate your Deployments to Astro` - is this correct? I was thinking that you need to set up authentication between Datadog and Astro, not the Deployments and Astro.

Author: Whoops sorry yeah that should read Datadog

## Export task logs to Datadog

> You can forward Airflow task logs from a Deployment to [Datadog](https://www.datadoghq.com/) using a Datadog API key. This allows you to view and manage task logs across all Deployments from a centralized observability plane.

Reviewer: The `Export metrics` section has a lot more "why" than this introduction piece.

### Prerequisites

> - Your Deployment must run Astro Runtime 9 (AWS) or 9.1 (Azure and GCP). See [Upgrade Astro Runtime](upgrade-runtime.md).

Reviewer: ```suggestion
- Your Deployment must run Astro Runtime 9 (AWS) or 9.1 (Azure and GCP) or later. See [Upgrade Astro Runtime](upgrade-runtime.md).
```

> - Your Deployment must be running Astro Runtime 9 (AWS) or 9.1 (Azure and GCP). See [Upgrade Astro Runtime](upgrade-runtime.md).

Reviewer: ```suggestion
- Your Deployment must run Astro Runtime 9 (AWS) or 9.1 (Azure and GCP). See [Upgrade Astro Runtime](upgrade-runtime.md).
```

### Setup

1. Create a new Datadog API key or copy an existing API key. See [API and Application Keys](https://docs.datadoghq.com/account_management/api-app-keys/).
2. Set the following [environment variable](environment-variables.md) on your Deployment:

    - **Key 1**: `DATADOG_API_KEY`
    - **Value 1**: Your Datadog API key.

    - **Key 2**: `ASTRO_DATADOG_TASK_LOGS_ENABLED`
    - **Value 2**: `true`

    Select the **Secret?** checkbox for `DATADOG_API_KEY`. This ensures that your Datadog API key is saved securely and is not available to Workspace users in plain text.

  :::info

>   By default, the Astro Datadog integration also sends a Deployment's [Airflow metrics](deployment-metrics.md#export-airflow-metrics-to-datadog) to Datadog. To send only task logs to Datadog, additionally set the following environment variable:

Reviewer: ```suggestion
  By default, the Astro Datadog integration also sends a Deployment's [Airflow metrics](deployment-metrics.md#export-airflow-metrics-to-datadog) to Datadog. To send only task logs to Datadog, add the following environment variable:
```

    - **Key**: `ASTRO_DATADOG_METRICS_DISABLED`
    - **Value**: `true`

  :::

3. (Optional) Set the following [environment variable](environment-variables.md) on your Deployment to send your logs to a specific [Datadog site](https://docs.datadoghq.com/getting_started/site/):

    - **Key**: `DATADOG_SITE`
    - **Value**: Your Datadog **Site Parameter**. For example, `datadoghq.com`.
   
4. (Optional) For Astro Runtime 9.2.0 and greater, set the following [environment variable](environment-variables.md) on your Deployment to [add specific tags to your logs](https://docs.datadoghq.com/getting_started/tagging/):

    - **Key**: `ASTRO_DATADOG_TASK_LOGS_TAGS`
    - **Value**: `<tag-key-1>:<tag-value-1>,<tag-key-2>:<tag-value-2>`

  By default, Astro uses the tags `source=astronomer` and `service=astronomer-task-logs`.

> 

Reviewer: ```suggestion
```

## Export Airflow metrics to Datadog

> Export over 40 Airflow metrics related to the state and performance of your Astro Deployment to [Datadog](https://www.datadoghq.com/) by adding a Datadog API key to your Deployment. These metrics include most information that is available in the Cloud UI, as well as additional metrics that Datadog automatically collects, including number of queued tasks, DAG processing time, and more. For a complete list of supported metrics, see [Data Collected](https://docs.datadoghq.com/integrations/airflow/?tab=host#data-collected) in the Datadog documentation.

Reviewer: Very thorough why! We might need to reference a couple of these points at the very beginning of the page, though.

Author: Is this a "why"? I thought this was just elaborating on the definition of a Datadog Airflow metric. But then again I'm not sure how I would add more "why" that isn't redundant with the overview section

> Export over 40 Airflow metrics related to the state and performance of your Astro Deployment to [Datadog](https://www.datadoghq.com/) by adding a Datadog API key to your Deployment. These metrics include most information that is available in the Cloud UI, as well as additional metrics that Datadog automatically collects, including number of queued tasks, DAG processing time, and more. For a complete list of supported metrics, see [Data Collected](https://docs.datadoghq.com/integrations/airflow/?tab=host#data-collected) in Datadog documentation.

Reviewer: ```suggestion
Export over 40 Airflow metrics related to the state and performance of your Astro Deployment to [Datadog](https://www.datadoghq.com/) by adding a Datadog API key to your Deployment. These metrics include most information that is available in the Cloud UI, as well as additional metrics that Datadog automatically collects, including number of queued tasks, DAG processing time, and more. For a complete list of supported metrics, see [Data Collected](https://docs.datadoghq.com/integrations/airflow/?tab=host#data-collected) in the Datadog documentation.
```
